### ISSUE TITLE
Curious crash reports from Google Play Console

### ISSUE THREAD

**Issue by hrydgard** [hrydgard/ppsspp#9718]
The most common is an empty call stack, most likely game crashes end up here. Maybe we should install an exception handler and check if the pointer is in "game space" and ignore the access and stop the game, otherwise pass through and allow the crash to happen so we still get reports.

Second most common:
```
backtrace:
  native: pc 0000000000038bd8  <unknown>
  native: pc 0000000000282353  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_Z18__KernelReSchedulePKc+54)
  native: pc 00000000fffffffd  <unknown>
```

Third:
```
signal 7 (SIGBUS), code 1 (BUS_ADRALN)
in _ZNK12VertexReader17ReadPosThroughZ16EPf

 native: pc 000000000036fdf0  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZNK12VertexReader17ReadPosThroughZ16EPf+31)
  native: pc 000000000036f449  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN13TransformUnit10ReadVertexER12VertexReader+40)
  native: pc 000000000036e82f  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN13TransformUnit15SubmitPrimitiveEPvS0_15GEPrimitiveTypeijPiP18SoftwareDrawEngine+1406)
  native: pc 000000000036d875  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN7SoftGPU9ExecuteOpEjj+2560)
  native: pc 000000000036ce57  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN7SoftGPU11FastRunLoopER11DisplayList+74)
  native: pc 00000000003601c9  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN9GPUCommon13InterpretListER11DisplayList+620)
  native: pc 0000000000360b09  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN9GPUCommon22ProcessDLQueueInternalEv+268)
  native: pc 000000000036ce03  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN7SoftGPU12ProcessEventE8GPUEvent+26)
  native: pc 000000000030420f  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN16ThreadEventQueueI12GPUInterface8GPUEvent12GPUEventTypeLS2_0ELS2_8ELS2_7EE14RunEventsUntilEy+414)
  native: pc 0000000000360ead  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN9GPUCommon14ProcessDLQueueEv+8)
  native: pc 000000000035fb0f  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN9GPUCommon11UpdateStallEij+118)
  native: pc 0000000000242f7f  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so
  native: pc 00000000002299cb  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_Z23CallSyscallWithoutFlagsPK11HLEFunction+14)
  native: pc 00000000000a6d58  <unknown>
Page
```
This one is weird because people really shouldn't be using the SoftGPU on Android. We might want to hide it in developer options on mobile devices which definitely don't have the grunt to use it.

Fourth:
```
backtrace:
  native: pc 0000000000014294  /system/lib/libc.so (strcmp+112)
  native: pc 00000000002116b9  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_ZN9ElfReader8LoadIntoEjb+228)
  native: pc 00000000fffffffd  <unknown>
```

Then there are a few more variants of third, and then we really go off into the weeds. Like:

```
backtrace:
  native: pc 00000000001c3408  /data/app/org.ppsspp.ppsspp-1/lib/arm64/libppsspp_jni.so (_Z15OpenSLWrap_InitPFiPsiEii+672)
  native: pc 00000000001c2ff8  /data/app/org.ppsspp.ppsspp-1/lib/arm64/libppsspp_jni.so (_Z19AndroidAudio_Resumev+76)
  native: pc 00000000004f2eb8  /data/app/org.ppsspp.ppsspp-1/oat/arm64/base.odex
```

```
backtrace:
  native: pc 00000000002e1a18  <unknown>
  native: pc 000000000027dca3  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (_Z21__KernelUnlockLwMutexI10PSPPointerI21NativeLwMutexWorkareaEEbT_Rj+82)
  native: pc 00000000fffffffd  <unknown>
```

```
java.lang.NullPointerException: 
  at android.app.LoadedApk.getAssets(LoadedApk.java:589)
  at android.app.LoadedApk.makeApplication(LoadedApk.java:682)
  at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6002)
  at android.app.ActivityThread.access$1700(ActivityThread.java:218)
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1795)
  at android.os.Handler.dispatchMessage(Handler.java:102)
  at android.os.Looper.loop(Looper.java:145)
  at android.app.ActivityThread.main(ActivityThread.java:6934)
  at java.lang.reflect.Method.invoke(Native Method:0)
  at java.lang.reflect.Method.invoke(Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
```

```
backtrace:
  native: pc 0000000000018186  /system/lib/libc.so (__memcpy_base+249)
  native: pc 00000000001afeb1  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so
  native: pc 0000000000003938  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so
```

**Comment by unknownbrackets** [comment]
We could set a flag in the ini and try to "detect" crashes.  Then if they start back up with the flag dirty, and any of these settings, we could suggest turning them off (maybe with a button):
 * fast memory
 * softgpu
 * cpu speed hacks
 * read framebuffers to memory
 * high upscale levels or render resolution (OOM)
 * multithreaded
 * cheats

Second - I feel like this means memory corruption or something breaking the idle threads.  Interesting that it's so common.

Third - also BUS_ADRALN.  Maybe something wrong in the 16 bit of float cases?  But agreed it doesn't make sense on mobile.

Fourth - https://github.com/hrydgard/ppsspp/blob/02d26279b73369a52f3906b159161f3524bfeb58/Core/ELF/ElfReader.cpp#L381 - https://github.com/hrydgard/ppsspp/blob/02d26279b73369a52f3906b159161f3524bfeb58/Core/ELF/ElfReader.cpp#L32
Looks like we don't have any safety check making sure this really points to a null terminated string.

-[Unknown]

**Comment by hrydgard** [comment]
I found third, it's a silly global u8 buffer that the compiler can align however it likes. Fixing that.

**Comment by hrydgard** [comment]
I don't know what more safety checks we could make around the name pointer in the ElfReader - ```if (name && !strcmp(name, ".rodata.sceModuleInfo")) {``` at least checks that it's not null before calling strcmp. We could check the range of nameOffset, I suppose.

**Comment by hrydgard** [comment]
Here's a savestate one that's not very common, but a bit worrying:

```
backtrace:
  native: pc 0000000000272bc8  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (Module::DoState(PointerWrap&)+843)
  native: pc 0000000000258f09  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (KernelObjectPool::DoState(PointerWrap&)+296)
  native: pc 0000000000258c15  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (__KernelDoState(PointerWrap&)+64)
  native: pc 00000000002eae67  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (SaveState::SaveStart::DoState(PointerWrap&)+302)
  native: pc 00000000002ef245  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (CChunkFileReader::Error CChunkFileReader::Load<SaveState::SaveStart>(std::string const&, char const*, SaveState::SaveStart&, std::string*)+80)
  native: pc 00000000002eeb05  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (SaveState::Process()+576)
  native: pc 00000000002f22f7  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (PSP_RunLoopUntil(unsigned long long)+66)
  native: pc 00000000003b3159  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (EmuScreen::render()+296)
  native: pc 0000000000449401  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (ScreenManager::render()+76)
  native: pc 00000000003ad0fd  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (NativeRender(GraphicsContext*)+1104)
  native: pc 00000000003a7e47  /data/app/org.ppsspp.ppsspp-1/lib/arm/libppsspp_jni.so (Java_org_ppsspp_ppsspp_NativeRenderer_displayRender+50)
  native: pc 0000000000004ea7  /data/dalvik-cache/arm/data@app@org.ppsspp.ppsspp-1@base.apk@classes.dex
```

**Comment by unknownbrackets** [comment]
I guess #9734 effectively blocks us from knowing if any of these fixes helped?

That said, since we released v1.4.2, should we keep this open?  Not sure how to remember to check back in after X weeks/months.

-[Unknown]

**Comment by hrydgard** [comment]
Yes. Still no response on my appeal - since they've left the Gold version up though I don't see how the appeal won't succeed. Anyway, if it takes too long I'll just re-upload with a new package name, losing the history and ranking though probably, sigh. Emulators are clearly still allowed.
